torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1757,6 +1758,65 @@ def no_sync(self) -> Generator:
                 )
                 m._require_backward_grad_sync = old_flag
 
+    @property
+    def params_with_grad(self) -> List[Parameter]:
+        """[p for p in self.parameters() if p.grad is not None]"""
```

Comment:
nit: comment is redundant because it is the next line itself. 

Maybe
"""
Recursively returns a list of all module parameters that have a gradient.
"""